images.CI/linux-and-win/azure-pipelines/image-generation.yml

```diff
@@ -71,28 +71,71 @@ jobs:
+
   - task: PowerShell@2
     displayName: 'Build VM'
     inputs:
       targetType: filePath
       filePath: ./images.CI/linux-and-win/build-image.ps1
-      arguments: -ResourcesNamePrefix $(Build.BuildId) `
-                        -ClientId $(CLIENT_ID) `
-                        -ClientSecret $(CLIENT_SECRET) `
-                        -TemplatePath $(TemplatePath) `
-                        -ResourceGroup $(AZURE_RESOURCE_GROUP) `
-                        -StorageAccount $(AZURE_STORAGE_ACCOUNT) `
-                        -SubscriptionId $(AZURE_SUBSCRIPTION) `
-                        -TenantId $(AZURE_TENANT) `
-                        -Location $(AZURE_LOCATION) `
-                        -VirtualNetworkName $(BUILD_AGENT_VNET_NAME) `
-                        -VirtualNetworkRG $(BUILD_AGENT_VNET_RESOURCE_GROUP) `
-                        -VirtualNetworkSubnet $(BUILD_AGENT_SUBNET_NAME)
+      arguments: -ClientId $(CLIENT_ID) `
+                 -ClientSecret $(CLIENT_SECRET) `
+                 -TemplatePath $(TemplatePath) `
+                 -ImageName "$(ManagedImageName)" `
+                 -ImageResourceGroupName $(AZURE_RESOURCE_GROUP) `
+                 -TempResourceGroupName "$(TempResourceGroupName)" `
+                 -SubscriptionId $(AZURE_SUBSCRIPTION) `
+                 -TenantId $(AZURE_TENANT) `
+                 -Location $(AZURE_LOCATION) `
+                 -VirtualNetworkName $(BUILD_AGENT_VNET_NAME) `
+                 -VirtualNetworkRG $(BUILD_AGENT_VNET_RESOURCE_GROUP) `
+                 -VirtualNetworkSubnet $(BUILD_AGENT_SUBNET_NAME)
 
     env:
       PACKER_LOG: 1
       PACKER_LOG_PATH: "$(Agent.TempDirectory)/packer-log.txt"
-  
+
+  - task: PowerShell@2
+    displayName: 'Convert managed image to VHD'
+    inputs:
+      targetType: filePath
+      filePath: ./images.CI/linux-and-win/convert-to-vhd.ps1
+      arguments: -SubscriptionId $(AZURE_SUBSCRIPTION) `
+                 -Location $(AZURE_LOCATION) `
+                 -ResourceGroupName $(AZURE_RESOURCE_GROUP) `
+                 -ManagedImageName "$(ManagedImageName)" `
+                 -GalleryName "github_imagegeneration_convert_to_vhd" `
+                 -GalleryImageSku "${{ parameters.image_type }}" `
+                 -GalleryImageVersion "0.0.$(Build.BuildId)" `
+                 -StorageAccountName $(AZURE_STORAGE_ACCOUNT) `
+                 -StorageAccountContainerName "images" `
+                 -VhdName "$(VhdName)" `
+                 -ClientId $(CLIENT_ID) `
+                 -ClientSecret $(CLIENT_SECRET) `
+                 -TenantId $(AZURE_TENANT)
+
+  - task: PowerShell@2
+    displayName: 'Remove managed image'
+    inputs:
+      targetType: 'inline'
+      script: |
+        az login --service-principal `
+          --username "$(CLIENT_ID)" `
+          --password "$(CLIENT_SECRET)" `
+          --tenant "$(AZURE_TENANT)" | Out-Null
+
+        az image delete `
```

Comment:
Should we move it to the "cleanup.ps1" script?

Reply:
I think it'd be better to move it to `convert-to-vhd.ps1` as it relates to image conversion while `cleanup.ps1` is about removing resources that were possibly left by packer.